src/main/java/gift/service/OrderService.java

```diff
@@ -0,0 +1,69 @@
+package gift.service;
+
+import gift.domain.Member;
+import gift.domain.Order;
+import gift.domain.ProductOption;
+import gift.dto.OrderRequest;
+import gift.dto.OrderResponse;
+import gift.exception.UnauthorizedException;
+import gift.repository.MemberRepository;
+import gift.repository.OrderRepository;
+import gift.repository.ProductOptionRepository;
+import gift.repository.WishRepository;
+import org.springframework.data.crossstore.ChangeSetPersister;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+public class OrderService {
+
+    private final MemberRepository memberRepository;
+    private final OrderRepository orderRepository;
+    private final ProductOptionRepository productOptionRepository;
+    private final WishRepository wishRepository;
+    private final KakaoMessageService kakaoMessageService;
+
+    public OrderService(MemberRepository memberRepository, OrderRepository orderRepository, ProductOptionRepository productOptionRepository, WishRepository wishRepository, KakaoMessageService kakaoMessageService) {
+        this.memberRepository = memberRepository;
+        this.orderRepository = orderRepository;
+        this.productOptionRepository = productOptionRepository;
+        this.wishRepository = wishRepository;
+        this.kakaoMessageService = kakaoMessageService;
+    }
+
+    @Transactional
+    public OrderResponse createOrder(Member loginMember, OrderRequest request) {
+
+        Member member = memberRepository.findById(loginMember.getId())
+                .orElseThrow(() -> new IllegalStateException("회원 정보를 찾을 수 없습니다."));
+
+        ProductOption option = productOptionRepository.findById(request.getOptionId())
+                .orElseThrow(() -> new IllegalArgumentException("해당 옵션이 존재하지 않습니다."));
+
+        option.subtractQuantity(request.getQuantity());
+        productOptionRepository.save(option);
+
+        wishRepository.deleteByMemberIdAndProductOptionId(member.getId(), option.getId());
+
+        Order order = new Order(member, option.getId(), request.getQuantity(), request.getMessage());
+        Order saved = orderRepository.save(order);
+
+        String message = """
+            주문이 완료되었어요!
+            - 옵션: %s
+            - 수량: %d개
+            - 메시지: %s
+            """.formatted(option.getName(), request.getQuantity(), request.getMessage());
+
+        kakaoMessageService.sendOrderMessageToMe(member.getAccessToken(), message);
```

Comment:
지금은 외부에서 메시지 템플릿을 만들어서 전달을해줘야하는데요. 
kakaoMessageService 객체에게 메시지를 보내는 역할을 온전히 위임해보면 어떨까요?

또한 잘 생각해보면 주문에 대한 정보이니 Order객체가 모두 잘 알고있는 정보이지 않을까요?  메시지 템플릿을 만들기위해서 order 객체에게 필요한 정보들을 꺼내서 사용할 수 있을 것 같아요. 

```suggestion
        kakaoMessageService.sendOrderMessageToMe(member.getAccessToken(), order);
```